cypress-io/cypress
# fix: discover CT community definitions in monorepos with hoisted dependencies

- Closes #25993

### Additional details


Right now when we detect CT community framework modules during the Launchpad setup, we only check the `projectRoot` `node_modules/` and then return. This doesn't work if you have a monorepo that hoists dependencies to the repository root. 

This PR adds logic to search from the project root to the repository root for community framework modules and return all of the ones that are found.

We use a variety of methods to determine whether or not we've found the repository root:
- Looking for platform-specific configuration files (eg `lerna.json`, `nx.json`, etc)
- Looking for a `.git/` directory (meaning that Git has been configured at this level, indicating a repository root)
- `package.json` having a `workspace` entry in it

This isn't a fool-proof way of determining the repository root but it should work for most projects. I basically copied what Vite does https://github.com/vitejs/vite/blob/0f6de4dcff783ec21fada47651d564cd5e2631b2/packages/vite/src/node/server/searchRoot.ts#L59

### Steps to test


Use a monorepo, eg https://github.com/lmiller1990/ct-monorepo-bug

- yarn install
- notice @lmiller1990/cypress-ct-solid-js is in the root node_modules
- cd packages/foo
- open yarn in global mode `yarn cypress:open`
- open `ct-monorepo-bug/packages/foo` in Cypress
- select CT
- verify that Solid shows up in the list as a community framework

### How has the user experience changed?


### PR Tasks


- [x] Have tests been added/updated? - I'm still working on this, will add another test for automatically testing monorepo support
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
